### ISSUE TITLE
[JIT] Alias Annotations for Native Schema List Ops Is Subtly Wrong

### ISSUE THREAD

**Issue by eellison** [pytorch/pytorch#47218]
## 🐛 Bug

Pytorch List Ops are schematized as `func: chunk(Tensor(a) self, int chunks, int dim=0) -> Tensor(a)[]`. Because we set all containers to store the wildcard element, the `Tensor(a)[]` annotation effectively is read as `Tensor(*)[]`. As a result, the input self is not read to be contained by the output list.

The correct annotation is: `func: chunk(Tensor(a -> *) self, int chunks, int dim=0) -> Tensor[]`. I tried updating the schemas and ran into some errors with internal codegen. I can get it to it if no one else wants to do it....
```
@torch.jit.script
def foo():
    x = torch.tensor([4, 4, 4, 4])
    for elem in x.chunk(4, dim=0):
        elem.add_(10)
    return x + 1
```
print(foo())
> tensor([5, 5, 5, 5])

cc: @suo

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
